src/Containers/Microsoft.NET.Build.Containers/Registry.cs

```diff
@@ -233,58 +229,28 @@ private async Task<HttpResponseMessage> GetBlobAsync(string repositoryName, stri
         return response;
     }
 
-    private static string? CheckIfRidExistsInGraph(RuntimeGraph graphForManifestList, IEnumerable<string> leafRids, string userRid) => leafRids.FirstOrDefault(leaf => graphForManifestList.AreCompatible(leaf, userRid));
-
-    private (IReadOnlyDictionary<string, PlatformSpecificManifest>, RuntimeGraph) ConstructRuntimeGraphForManifestList(ManifestListV2 manifestList, RuntimeGraph dotnetRuntimeGraph)
+    private static IReadOnlyDictionary<string, PlatformSpecificManifest> ConstructRuntimeGraphForManifestList(ManifestListV2 manifestList)
     {
         var ridDict = new Dictionary<string, PlatformSpecificManifest>();
-        var runtimeDescriptionSet = new HashSet<RuntimeDescription>();
-        foreach (var manifest in manifestList.manifests) {
+        foreach (PlatformSpecificManifest manifest in manifestList.manifests) {
             if (CreateRidForPlatform(manifest.platform) is { } rid)
             {
-                if (ridDict.TryAdd(rid, manifest)) {
-                    AddRidAndDescendantsToSet(runtimeDescriptionSet, rid, dotnetRuntimeGraph);
-                }
+                ridDict.TryAdd(rid, manifest);
             }
         }
-
-        var graph = new RuntimeGraph(runtimeDescriptionSet);
-        return (ridDict, graph);
+        return ridDict;
     }
 
-    private static string? CreateRidForPlatform(PlatformInformation platform)
+    private static string? CreateRidForPlatform(PlatformInformation platform, bool isMuslBased = false)
```

Comment:
Maybe this `isMuslBased` can also be removed when the rid graph is used again?
(The graph maps `linux-musl-x64` to `linux-x64` which corresponds to an image platform.)

Reply:
You can use the `NETCoreSdkPortableRuntimeIdentifier` property and check if it starts with `linux-musl-`.

Reply:
Ignore that. This is the target rid.

Either it's `linux-musl-{arch}` as the portable rid.
Or if it's not portable, you can run it through the runtime graph and see if matches `linux-musl`.

Reply:
Yeah, that's a big part of it. I was planning to leave it for a different PR to do a full treatment, it just felt wrong to not acknowledge the concept in this PR

Reply:
I don't think the registry needs to know about `isMuslBased`.

Like a non-portable rid, the runtime graph will map `linux-musl` to the most compatible rid provided by a container image (which is: `linux`).

Reply:
The problem is that images don't describe if they are musl based.

The sdk could know for some images (in particular: the Microsoft ones) that they are musl-based based on their name.

Reply:
Yeah, that's the kind of determination I was thinking about hooking up later. For the MS images we know a family called `alpine` exists so if the user asked for a musl RID we could auto-insert that. Anything else we probably can't rely on - we just have to accept that a user knows what they are doing if the eject out of inference.

Reply:
Yes. That could be handled by the `ComputeDotnetBaseImageTag` Task if we pass it the `RuntimeIdentifier`.